packages/lexical-utils/src/index.ts

```diff
@@ -511,3 +511,29 @@ export function isHTMLElement(x: Node | EventTarget): x is HTMLElement {
   // @ts-ignore-next-line - strict check on nodeType here should filter out non-Element EventTarget implementors
   return x.nodeType === 1;
 }
+
+type IsInstanceOfSupportedType = Event | Node;
+export type ConstructorOf<C> = {new (...args: never[]): C};
+
+/**
+ * @param object = The instance of the global type
+ * @param globalClass = The class of the global type
+ * @returns Whether the object is an instance of the global class, ignoring the difference across window (e.g. different iframs)
+ */
+export function isInstanceOfCrossWindow<T extends IsInstanceOfSupportedType>(
+  object: unknown,
+  globalClass: ConstructorOf<T>,
+): object is T {
```

Comment:
```suggestion
): boolean {
```

This is not correct given that we're just checking equality. The object Class might not be the same as the Class you're comparing against